compiler/driver.ml

```diff
@@ -202,8 +202,21 @@ let driver source_file (options : Cli.options) : int =
           end
           else prgm
         in
+        (* Cli.debug_print (Format.asprintf "Typechecking results :@\n%a"
+           (Print.typ prgm.decl_ctx) typ); *)
         match backend with
         | `Typecheck ->
+          Cli.debug_print "Typechecking again...";
```

Comment:
If there's no bug in the compiler this re-typechecking should be transparent to the user. And `Typecheck`still fails as soon as possible : if there's a typing error it will be caught in `Scopelang` and abort. But when deving on the compiler the fact that `Typecheck` does `Dcalc` typechecking is really helpful. Otherwise I found myself to use `Interpret` instead of `Typecheck` just to have that double typechecking in my dev workflow.

Reply:
Fair enough !
At some point we should have options to toggle these checks on or off, but I agree we're not quite there yet :)